src/Shell32.Desktop/Shell32+KNOWNFOLDERID.cs

```diff
@@ -0,0 +1,526 @@
+            public static readonly Guid FOLDERID_System = new Guid("1AC14E77-02E7-4E5D-B744-2EB1AE5198B7");
+
+            /// <summary>System32</summary>
+            /// <remarks>Maps to %windir%\system32</remarks>
+            public static readonly Guid FOLDERID_SystemX86 = new Guid("D65231B0-B2F1-4857-A4CE-A8E7C6EA7D27");
+
+            /// <summary>Templates</summary>
+            /// <remarks>Maps to %APPDATA%\Microsoft\Windows\Templates</remarks>
+            public static readonly Guid FOLDERID_Templates = new Guid("A63293E8-664E-48DB-A079-DF759E0509F7");
+
+            /// <summary>Not used</summary>
+            /// <remarks>Not used in Windows Vista. Unsupported as of Windows 7.</remarks>
+            public static readonly Guid FOLDERID_TreeProperties = new Guid("5b3749ad-b49f-49c1-83eb-15370fbd4882");
+
+            /// <summary>User Pinned</summary>
+            /// <remarks>Maps to %APPDATA%\Microsoft\Internet Explorer\Quick Launch\User Pinned</remarks>
+            public static readonly Guid FOLDERID_UserPinned = new Guid("9E3995AB-1F9C-4F13-B827-48B24B6C7174");
+
+            /// <summary>Users</summary>
+            /// <remarks>Maps to %SystemDrive%\Users</remarks>
+            public static readonly Guid FOLDERID_UserProfiles = new Guid("0762D272-C50A-4BB0-A382-697DCD729B80");
+
+            /// <summary>Programs</summary>
+            /// <remarks>Maps to %LOCALAPPDATA%\Programs</remarks>
+            public static readonly Guid FOLDERID_UserProgramFiles = new Guid("5CD7AEE2-2219-4A67-B85D-6C9CE15660CB");
+
+            /// <summary>Programs</summary>
+            /// <remarks>Maps to %LOCALAPPDATA%\Programs\Common</remarks>
+            public static readonly Guid FOLDERID_UserProgramFilesCommon = new Guid("BCBD3057-CA5C-4622-B42D-BC56DB0AE516");
+
+            /// <summary>The user's full name entered when the user account was created.</summary>
+            /// <remarks>virtual folder</remarks>
+            public static readonly Guid FOLDERID_UsersFiles = new Guid("f3ce0f7c-4901-4acc-8648-d5d44b04ef8f");
+
+            /// <summary>Libraries</summary>
+            /// <remarks>virtual folder</remarks>
+            public static readonly Guid FOLDERID_UsersLibraries = new Guid("A302545D-DEFF-464b-ABE8-61C8648D939B");
+
+            /// <summary>Videos</summary>
+            /// <remarks>Maps to %USERPROFILE%\Videos</remarks>
+            public static readonly Guid FOLDERID_Videos = new Guid("18989B1D-99B5-455B-841C-AB7C74E4DDFC");
+
+            /// <summary>Videos</summary>
+            /// <remarks>Maps to %APPDATA%\Microsoft\Windows\Libraries\Videos.library-ms</remarks>
+            public static readonly Guid FOLDERID_VideosLibrary = new Guid("491E922F-5643-4AF4-A7EB-4E7A138D8174");
+
+            /// <summary>Windows</summary>
+            /// <remarks>Maps to %windir%</remarks>
+            public static readonly Guid FOLDERID_Windows = new Guid("F38BF404-1D43-42F2-9305-67DE0B28FC23");
```

Comment:
Tip: Constructing `Guid` objects based on integers rather than parsing strings [is 42X faster](https://twitter.com/aarnott/status/701653964823990273). For a large set of static fields that could help startup perf for a consuming application. 
Not necessary for this PR, but sometime I'd like to write a tool that will convert all these to their faster variety.